modules/library/src/main/scala/zio/elasticsearch/ElasticAggregation.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticAggregationType.{Multiple, Terms}
+import zio.elasticsearch.ElasticPrimitive._
+import zio.elasticsearch.OtherAggregation.WithOtherAggregation
+import zio.elasticsearch.SubAggregation.WithSubAggregation
+import zio.json.ast.Json
+import zio.json.ast.Json.Obj
+
+sealed trait ElasticAggregation[EAT <: ElasticAggregationType] { self =>
+
+  def paramsToJson: Json
+
+  final def sub(aggregation: ElasticAggregation[_ <: SingleElasticAggregationType])(implicit
+    wsa: WithSubAggregation[EAT]
+  ): ElasticAggregation[EAT] =
+    wsa.sub(aggregation = self, subAggregation = aggregation)
+
+  final def toJson: Json = Obj("aggs" -> paramsToJson)
+
+  final def `with`(aggregation: ElasticAggregation[_ <: SingleElasticAggregationType])(implicit
```

Comment:
@arnoldlacko should we use `withAggregation`?

Reply:
Yes, or I wonder if `withAgg` might be better to not make the querie longer than necessary  🤔

Reply:
I far as I remember Elastic also uses `aggs` so it should be clear in enough what it means

Reply:
I'm fine with either